torch/_dynamo/compiled_autograd.py

```diff
@@ -153,6 +157,92 @@ def begin_capture(
         self.stack.enter_context(disable_autocast_cache())
         return inputs, sizes, scalars
 
+    def proxy_call_lambda(
+        self,
+        idx,
+        inputs,
+        output_metadatas: List[Optional[Any]],
+    ):
+        with disable_proxy_modes_tracing():
+            # create fake Tensors
+            grad_ins: List[Optional[torch.Tensor]] = []
+            for output_metadata in output_metadatas:
+                if output_metadata is None:
+                    continue
+
+                layout, device, dtype, size = output_metadata
+                grad_ins.append(
+                    torch.empty(size=size, dtype=dtype, layout=layout, device=device)
+                )
+
+            global next_op
+
+            @torch.library.custom_op(  # type: ignore[misc]
+                f"compiled_autograd::cpp_node_op_{next_op}", mutates_args=()
+            )
+            def cpp_node_op_i(
```

Comment:
@jansel this is related to what we were talking about. Since the function behaves differently when wrapped in a custom op vs not in a custom op, the behavior may end up diverging from eager mode

Reply:
So having a custom_op that behaves more like a normal funciton would fix that?

Reply:
It depends on how technically feasible "a custom_op that behaves more like a normal function" is, but it would certainly help (if we ignore Tensor subclasses for a second). With Tensor subclasses I'm not sure anymore, and @xmfan tells me that Tensor subclasses are important for composing with some of the distributed use cases?

Reply:
How do C++ autograd nodes handle tensor subclasses?  Naively I would expect it not to work, unless the custom op consists entirely of other aten ops (in which case we can just trace it and don't need to capture it).

Reply:
Usually the C++ autograd node consists of other aten ops. But note that we can't always "just trace it", because it may lead to specialization on constants (if the C++ autograd node saves a float, then we end up specializing on it).

Reply:
Won't turning it into an opaque op have the same problem?  The float is still saved on the object...

Reply:
the graph just sees `call_custom_op(id)`, the float is captured in the lambda binded to `custom_op_impls[id]`